src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1112,6 +1123,148 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
         public bool HasImplicitConversion(ITypeSymbol fromType, ITypeSymbol toType)
             => fromType != null && toType != null && this.ClassifyCommonConversion(fromType, toType).IsImplicit;
 
+        /// <summary>
+        /// Checks if <paramref name="symbol"/> is accessible from within <paramref name="within"/>. An optional qualifier of type
+        /// <paramref name="throughType"/> is used to resolve protected access for instance members. All symbols are
+        /// required to be from this compilation or some assembly referenced (<see cref="References"/>) by this
+        /// compilation. <paramref name="within"/> is required to be an <see cref="ITypeSymbol"/> or <see cref="IAssemblySymbol"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>Submissions can reference symbols from previous submissions and their referenced assemblies, even
+        /// though those references are missing from <see cref="References"/>.
+        /// See https://github.com/dotnet/roslyn/issues/27356.
+        /// This implementation works around that by permitting symbols from previous submissions as well.</para>
```

Comment:
>This implementation works around that by permitting symbols from previous submissions as well. [](start = 12, length = 94)

I don't thin this information and the reference to [https://github.com/dotnet/roslyn/issues/27356](https://github.com/dotnet/roslyn/issues/27356) belongs in the doc comment.

Reply:
I'll check with the API design review on that question.

Reply:
API design review has approved the specification based on `Compilation.References`, with this caveat in place.

---
In reply to: [193173129](https://github.com/dotnet/roslyn/pull/27312#discussion_r193173129) [](ancestors = 193173129)